internal/events/aggregator_rewind.go

```diff
@@ -283,3 +293,27 @@ func (rw *rewinder) getRewindsForBlobs(ctx context.Context, newHashes []driver.V
 
 	return nil
 }
+
+func (rw *rewinder) getRewindsForDIDs(ctx context.Context, dids []driver.Value, batchIDs map[fftypes.UUID]bool) error {
+
+	// We need to find all pending messages, that are authored by this DID
+	fb := database.MessageQueryFactory.NewFilter(ctx)
+	filter := fb.And(
+		fb.Eq("state", fftypes.MessageStatePending),
+		fb.In("author", dids),
+	)
+	records, err := rw.database.GetMessageIDs(ctx, filter)
```

Comment:
Is there a risk here that this becomes a large query and needs to be paginated?

Reply:
I did consider that, and determined the answer should be no (for now):
- We are only querying the id+sequence, so the in-memory size is very small per record
- We are only querying things from a given author that has just become confirmed, so we should be dealing with a small number of records